### ISSUE TITLE
Using GPT-4?

### ISSUE THREAD

**Issue by morethanx** [brianpetro/obsidian-smart-connections#57]
Hi,
I'm having so much fun using your chat plugin to search my notes better and get insights. I was just wondering if I was able to change the app for myself to use gpt-4-0314, and when I went through your code to see if I can adapt it for myself without having much experience in coding I found this part:

const DEFAULT_SETTINGS = {
  api_key: "",
  file_exclusions: "",
  folder_exclusions: "",
  header_exclusions: "",
  path_only: "",
  show_full_path: false,
  expanded_view: true,
  log_render: false,
  log_render_files: false,
  skip_sections: false,
  smart_chat_model: ==**"gpt-3.5-turbo",**==
  results_count: 30,
  view_open: true,
  version: "",

At smart-chat-model, it says 3.5. Doesn't that mean, it's always 3.5, even when I have a GPT-4 API? Sorry if this is a stupid question, just a newbie here. :) and again, thanks!

**Comment by brianpetro** [comment]
> I'm having so much fun using your chat plugin to search my notes better and get insights.

I love to hear that. Thanks for letting me know!

You can change the model to GPT-4  in the Smart Connections settings.

If you don't see the option, make sure the app is updated using the "Check for updates" button in the "Community plugins" section of Obsidian.

I hope that helps!

Brian🌴

**Comment by morethanx** [comment]
Thanks for such a quick response. I already changed it to gpt-4, but  it says it doesn't know about any gpt-4 because it only has data up until March 21.

**Comment by brianpetro** [comment]
@morethanx can you send a screenshot?

**Comment by morethanx** [comment]
Im not at my computer now, but can certainly do that later.

**Comment by morethanx** [comment]
Hi there, sorry having taken so long. I attached some screenshot. Somehow it is not working at all anymore now. I asked it which GPT it is and it's been working on responding to this for ages (and still didn't come up with a response. The first time this happened was March 17, you see this in my usage history. I closed obsidian at one point because it just didn't stop, but since then it hasn't been working properly anymore. Would you delete and reinstall?

 <img width="1184" alt="Screenshot 2023-03-23 at 12 36 39" src="https://user-images.githubusercontent.com/96887658/227192918-ce4aeacc-2e4d-4a39-a9c7-7eae4e4323b6.png">
<img width="816" alt="Screenshot 2023-03-23 at 12 39 15" src="https://user-images.githubusercontent.com/96887658/227192930-2832a2bc-0161-4e53-8a63-591f679b9d7a.png">
<img width="483" alt="Screenshot 2023-03-23 at 12 40 28" src="https://user-images.githubusercontent.com/96887658/227192936-2ff3a7e1-8fc1-43c7-993b-22a6035c09e2.png">

**Comment by morethanx** [comment]
hi, I reinstalled it and asked it which version it is. It works again, but it still says GPT-4? Is that just a glitch? I double checked, I put it on GPT-4 in the settings.

<img width="479" alt="Screenshot 2023-03-23 at 12 59 39" src="https://user-images.githubusercontent.com/96887658/227197371-a36229a4-bbff-44ce-99ab-0fc24626eb7b.png">

**Comment by brianpetro** [comment]
@morethanx I'm glad that we seem to be making progress 😊

If you're trying to verify which model is being used, asking the chat isn't a reliable method. Instead, look at your [OpenAI Account Usage](https://platform.openai.com/account/usage) for a more accurate picture.

![image](https://user-images.githubusercontent.com/1886014/227204178-eeac20ad-7809-495f-9618-65cede142e39.png)

Additionally, I received some errors from the GPT-4 API this morning. This led me to improve the error handling in the latest version, `1.4.1`.

Let me know if that helps clear some things up for you.

Thanks for following up,
Brian🌴

**Comment by morethanx** [comment]
Thank you! It still tells me that it's only using data until Sep 21. <img width="347" alt="Screenshot 2023-03-23 at 18 36 07" src="https://user-images.githubusercontent.com/96887658/227291716-c773b7d6-43ad-49e4-a648-0c51b3a1047c.png">

**Comment by brianpetro** [comment]
@morethanx asking an LLM model about itself is not effective. It only knows what it has been told via prompts. The September 2021 date simply comes from a prompt that is included behind the scenes by OpenAI.

**Comment by morethanx** [comment]
alright. I'll stop grilling it then. :) Thank you!

**Comment by gillchristian** [comment]
Hi, amazing plugin! :fire: 

Reusing this issue since it's related, I just installed it on a few Vaults, enabled the billing on the OpenAI account and set the model to GPT4 but I'm still getting errors when trying to chat.

The request to this endpoint `https://api.openai.com/v1/chat/completions` is producing the following error:

```
{
    "error": {
        "message": "The model: `gpt-4` does not exist",
        "type": "invalid_request_error",
        "param": null,
        "code": "model_not_found"
    }
}
```

Here's the usage in OpenAI

![image](https://user-images.githubusercontent.com/8309423/228478187-8e818e91-5d83-49af-8bce-cda1c83ab4b9.png)

I tried the hard refresh and it didn't work.

What did work is to switch to the GPT3 model. Could I be missing something on OpenAI side? Maybe GPT4 is not yet enabled there?
